src/lib/core/ripple/ripple-renderer.ts

```diff
@@ -143,6 +143,8 @@ export class RippleRenderer {
 
     rippleDiv.addEventListener('transitionend',
         (event: TransitionEvent) => transitionEndCallback(ripple, event));
+    // Remove the ripple after transitions (fade-in, transform, fade-out)
+    setTimeout(() => this.removeRippleFromDom(ripple.rippleElement), fadeInSeconds * 3 * 1000);
```

Comment:
Sometimes the element is only `hidden`, or `display: none`. In this case `ngOnDestroy` is never called. 

If we use `*ngIf` to remove the element, the ripple will be removed when it's removed from the DOM tree.